File: common/src/main/java/com/ssblur/scriptor/effect/ScriptorEffects.java
package com.ssblur.scriptor.effect;

import com.ssblur.scriptor.ScriptorMod;
import dev.architectury.registry.registries.DeferredRegister;
import dev.architectury.registry.registries.RegistrySupplier;
import net.minecraft.core.Holder;
import net.minecraft.core.registries.Registries;
import net.minecraft.world.effect.MobEffect;

@SuppressWarnings("unused")
public class ScriptorEffects {
  public static final DeferredRegister<MobEffect> EFFECTS = DeferredRegister.create(ScriptorMod.MOD_ID, Registries.MOB_EFFECT);
  public static final RegistrySupplier<MobEffect> HOARSE = EFFECTS.register("hoarse", MuteStatusEffect::new);
  public static final RegistrySupplier<MobEffect> MUTE = EFFECTS.register("mute", MuteStatusEffect::new);
  public static final RegistrySupplier<MobEffect> PHASING = EFFECTS.register("phasing", PhasingStatusEffect::new);
  public static final RegistrySupplier<MobEffect> WILD_PHASING = EFFECTS.register("wild_phasing", WildPhasingStatusEffect::new);
  public static final RegistrySupplier<MobEffect> SILVER_TONGUE = EFFECTS.register("silver_tongue", () -> new EmpoweredStatusEffect(0.8f));

  public static void register() {
    EFFECTS.register();
  }

  public static Holder<MobEffect> get(RegistrySupplier<MobEffect> effect) {
    return EFFECTS.getRegistrar().getHolder(effect.getId());
  }
}

File: common/src/main/kotlin/com/ssblur/scriptor/effect/ScriptorEffects.kt
package com.ssblur.scriptor.effect

import com.ssblur.scriptor.ScriptorMod
import com.ssblur.scriptor.ScriptorMod.registerEffect
import com.ssblur.unfocused.constructors.Effect
import com.ssblur.unfocused.registry.RegistrySupplier
import dev.architectury.registry.registries.DeferredRegister
import net.minecraft.core.registries.Registries
import net.minecraft.world.effect.MobEffect
import net.minecraft.world.effect.MobEffectCategory

@Suppress("unused")
object ScriptorEffects {
    val EFFECTS: DeferredRegister<MobEffect> = DeferredRegister.create(ScriptorMod.MOD_ID, Registries.MOB_EFFECT)
    val HOARSE: RegistrySupplier<MobEffect> = registerEffect("hoarse") { Effect(MobEffectCategory.HARMFUL, 0xcc00cc) {} }
    val MUTE: RegistrySupplier<MobEffect> = registerEffect("mute") { Effect(MobEffectCategory.HARMFUL, 0xcc00cc) {} }
    val PHASING: RegistrySupplier<MobEffect> = registerEffect("phasing") { PhasingStatusEffect() }
    val WILD_PHASING: RegistrySupplier<MobEffect> = registerEffect("wild_phasing") { WildPhasingStatusEffect() }
    val SILVER_TONGUE: RegistrySupplier<MobEffect> = registerEffect("silver_tongue") { EmpoweredStatusEffect(0.8f) }

    fun register() = EFFECTS.register()
}